### ISSUE TITLE
Adding Polyloss to `torch`

### ISSUE THREAD

**Issue by tanmoyio** [pytorch/pytorch#76732]
### 🚀 The feature, motivation and pitch

# POLYLOSS: A POLYNOMIAL EXPANSION PERSPEC- TIVE OF CLASSIFICATION LOSS FUNCTIONS
(Published as a conference paper at ICLR 2022)

This loss function implementation won't be difficult as we can use builtin crossentropy loss or focal loss to implement it. This loss function has overperformed other existing loss functions in some benchmark experiments. 

Here is a code snippet
```python3
# cross entropy as pollyloss
def poly1_cross_entropy(logits, labels, epsilon=1.0):
# pt, CE, and Poly1 have shape [batch].
pt = tf.reduce_sum(labels * tf.nn.softmax(logits), axis=-1) CE = tf.nn.softmax_cross_entropy_with_logits(labels, logits) Poly1 = CE + epsilon * (1 - pt)
return Poly1

# focal loss as pollyloss
def poly1_focal_loss(logits, labels, epsilon=1.0, gamma=2.0): # p, pt, FL, and Poly1 have shape [batch, num of classes]. p = tf.math.sigmoid(logits)
pt = labels * p + (1 - labels) * (1 - p)
    FL = focal_loss(pt, gamma)
    Poly1 = FL + epsilon * tf.math.pow(1 - pt, gamma + 1)
return Poly1
```

I would like to contribute to add this feature into `torch`

**Comment by jbschlosser** [comment]
Hey @tanmoyio, thanks for the request! Note that we have a fairly strict policy governing acceptance of new modules within PyTorch core. Details can be found [here](https://github.com/pytorch/pytorch/wiki/Developer-FAQ#i-have-a-new-function-or-feature-id-like-to-add-to-pytorch-should-it-be-in-pytorch-core-or-a-library-like-torchvision), but the TL;DR is that we recommend implementing new modules in a separate repo outside of PyTorch core until they have gained widespread acceptance in the community. This helps keep module maintenance manageable with our limited bandwidth. We can revisit the addition at a later time if this request becomes very popular. Thanks again for your interest in contributing!
